server/src/lib/jobs/resolve.cpp

```diff
@@ -396,6 +396,46 @@ static std::string byHostedBy(const Group::Condition& cond)
 
 static std::string byGroupId(fty::db::Connection& conn, const Group::Condition& cond);
 
+static std::string byTag(const Group::Condition& cond)
+{
+    auto condNot = cond;
+    if (cond.op == Group::ConditionOp::IsNot) {
+        condNot.op = Group::ConditionOp::Is;
+    } else if (cond.op == Group::ConditionOp::DoesNotContain) {
+        condNot.op = Group::ConditionOp::Contains;
+    }
+
+    static std::string sql = R"(
+        SELECT DISTINCT r.id_asset_element 
+        FROM t_bios_asset_element_tag_relation AS r 
+        INNER JOIN t_bios_tag AS tag 
+        ON r.id_tag = tag.id_tag 
+        WHERE tag.name {op} '{val}'
+    )";
+
+    if (condNot.op != cond.op) {
+        // clang-format off
+        std::string sqlFormat = fmt::format(sql,
+            "op"_a  = op(condNot),
+            "val"_a = value(condNot)
+        );
+
+        return fmt::format(R"( 
+            SELECT id_asset_element
+            FROM t_bios_asset_element
+            WHERE id_asset_element NOT IN ({})
+        )", sqlFormat);
+        // clang-format on
+    }
+
+    // clang-format off
+    return fmt::format(sql,
+        "op"_a  = op(cond),
+        "val"_a = value(cond)
+    );
+    // clang-format on
+}
```

Comment:
INFO... this would look better.
```cpp
   static std::string sql =  R"(
        SELECT DISTINCT r.id_asset_element 
        FROM t_bios_asset_element_tag_relation AS r 
        INNER JOIN t_bios_tag AS tag 
        ON r.id_tag = tag.id_tag 
        WHERE tag.name {op} '{val}'
    )";
    
    if (cond.op == Group::ConditionOp::IsNot || cond.op == Group::ConditionOp::DoesNotContain) {
        // reverse condition
        auto condNot = cond;
        if (cond.op == Group::ConditionOp::IsNot) {
           condNot.op = Group::ConditionOp::Is;
        } else if (cond.op == Group::ConditionOp::DoesNotContain) {
           condNot.op = Group::ConditionOp::Contains;
        }
        
        std::string sqlFormat = fmt::format(sql,
            "op"_a  = op(condNot),
            "val"_a = value(condNot)
        );

        return fmt::format(R"( 
            SELECT id_asset_element
            FROM t_bios_asset_element
            WHERE id_asset_element NOT IN ({})
        )", sqlFormat);
    } else {
        return fmt::format(sql,
          "op"_a  = op(cond),
          "val"_a = value(cond)
        );
    }
```